updating web components-related firefox support information, to make …

chrisdavidmills opened the pull request:
As per https://bugzilla.mozilla.org/show_bug.cgi?id=1503019#c16

Basically, I have removed a bunch of data about prefs that I thought was just making things confusing, and changed support versions for the latest pref info so that it no longer looks like it is supported behind a pref later than it is enabled by default (was really confusing)

ExE-Boss commented:
Also, if you’re gonna do this, at least change `version_added` to include the older pref data which you are removing.

chrisdavidmills commented:
Hi @ddbeck ! So you are correct on all counts here.

As per the one point of contention — the initial pref information was correct (the first set that I deleted), but I felt that it made the data unnecessarily uncomplicated. The Google Chrome folks have mentioned a number of times that they feel it is completely pointless including pref information after the feature has been enabled. Because noone cares or will find it useful.

However, I guess this is a conversation to have with a wider audience, not a decision for me to make on the fly. If you'd rather I added it back in, I will do.

ddbeck commented:
Sorry, @Elchi3 I opened #3318 before I saw your comment. I think we're on the same page that a decision is needed, but let me know if I should move my commentary to Discourse.

@chrisdavidmills in the interest of getting this PR merged, would you mind restoring the data you removed? That let us defer the decision about removing flagged data.

chrisdavidmills commented:
@ddbeck OK, done.
